npm-shrinkwrap.json

```diff
@@ -17903,12 +17891,14 @@
             "balanced-match": {
               "version": "1.0.0",
               "bundled": true,
-              "dev": true
+              "dev": true,
+              "optional": true
             },
             "brace-expansion": {
               "version": "1.1.11",
               "bundled": true,
               "dev": true,
+              "optional": true,
```

Comment:
@rarkins any idea why these `optional` flags are being set? We don't see them when generating shrinkwrap on development (OSX).

Reply:
If you take this branch and run `npm install` on it from OSX, does *that* modify the `optional` flags back again? I'd hope that once they're there, at least then they don't get removed.

The `npm` CLI team told me that they don't intend to implement a flag like `npm install --from-lock-file` and that using `npm ci` whenever you *don't* plan to update the lock file is the recommended approach.

Reply:
> If you take this branch and run npm install on it from OSX, does that modify the optional flags back again? I'd hope that once they're there, at least then they don't get removed.

Nope, it rewrites them. :(

Looks like we could use `npm install --no-save` in the install script to avoid it.